adapters/ix/ix.go

```diff
@@ -1,267 +1,189 @@
 package ix
 
 import (
-	"bytes"
-	"context"
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"net/http"
 
-	"github.com/prebid/prebid-server/openrtb_ext"
-
-	"github.com/prebid/prebid-server/pbs"
-
-	"golang.org/x/net/context/ctxhttp"
-
+	"github.com/golang/glog"
 	"github.com/mxmCherry/openrtb"
 	"github.com/prebid/prebid-server/adapters"
 	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
 )
 
-// maximum number of bid requests
-const requestLimit = 20
+var maxRequests = 20 // it's not a const for the unit test convenience
 
 type IxAdapter struct {
 	http *adapters.HTTPAdapter
 	URI  string
 }
 
-// Name is used for cookies and such
-func (a *IxAdapter) Name() string {
-	return string(openrtb_ext.BidderIx)
-}
-
-func (a *IxAdapter) SkipNoCookies() bool {
-	return false
-}
-
-type indexParams struct {
-	SiteID string `json:"siteId"`
-}
-
-type ixBidResult struct {
-	Request      *callOneObject
-	StatusCode   int
-	ResponseBody string
-	Bid          *pbs.PBSBid
-	Error        error
-}
-
-type callOneObject struct {
-	requestJSON bytes.Buffer
-	width       uint64
-	height      uint64
-}
+func (a *IxAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	logAsJSON("bid-request: %s", request)
 
-func isValidIXSize(f openrtb.Format, s [2]uint64) bool {
-	if f.W != s[0] || f.H != s[1] {
-		return false
+	nImp := len(request.Imp)
```

Comment:
You don't need this check. The core framework won't call `MakeRequests` is there are no impressions.

Reply:
I've realized that removing this check will not affect the safety of this function regardless of the core framework guaranties.